Content/Guides/syslog-ng-guide-admin/parser-regexp-options.htm

```diff
@@ -7,7 +7,7 @@
     <body name="parser-regexp-options">
         <h1 name="parser-regexp-options">Options of Regular expression parsers</h1>
         <p>The Regular expression parser has the following options.</p>
-        <MadCap:snippetBlock src="../shared/chunk/option-source-flags.htm">
+        <MadCap:snippetBlock src="regexp-flags-options-pcre.htm">
```

Comment:
This doesn't look good in the built doc (I was expecting this as a complete html page with h1 headers are included).

Also, I guess when documentation snippets are reused/included they should be in the `shared` folder, right?

Reply:
Hi, what would be your end goal? To add this section as a subchapter of **Options of Regular expression parsers**?
If yes, and this is a separate chapter with h1 headers etc, instead of adding it to a content file, you must add it to the relevant location of the `Project/TOCs/syslog-ng-guide-admin.fltoc` file.

As a rule, the `.htm` files cannot be used as snippets inside another `.htm` file. Only .flsnp snippets (that are just reusable chunks of information, e.g. a few `<p>` sentences without `<h1>` headers that should appear in several different chapters). 
There are still examples of using `.htm` files as snippets inside another `.htm` file in the OSE documentation, but we plan to get rid of these in all document sets if we successfully decide on an adequate alternative technical solution for it.

For example, from line 1325, instead of
```
<TocEntry
      Title="[%=System.LinkedTitle%]"
      Link="/Content/Guides/syslog-ng-guide-admin/parser-regexp.htm" xmlns:MadCap="http://www.madcapsoftware.com/Schemas/MadCap.xsd">
      <TocEntry
        Title="[%=System.LinkedTitle%]"
        Link="/Content/Guides/syslog-ng-guide-admin/parser-regexp-options.htm" xmlns:MadCap="http://www.madcapsoftware.com/Schemas/MadCap.xsd" />
</TocEntry>
```
it would look like this:
```
<TocEntry
      Title="[%=System.LinkedTitle%]"
      Link="/Content/Guides/syslog-ng-guide-admin/parser-regexp.htm" xmlns:MadCap="http://www.madcapsoftware.com/Schemas/MadCap.xsd">
      <TocEntry
        Title="[%=System.LinkedTitle%]"
        Link="/Content/Guides/syslog-ng-guide-admin/parser-regexp-options.htm" xmlns:MadCap="http://www.madcapsoftware.com/Schemas/MadCap.xsd" />
      <TocEntry
        Title="[%=System.LinkedTitle%]"
        Link="/Content/Guides/syslog-ng-guide-admin/regexp-flags-options-pcre.htm" xmlns:MadCap="http://www.madcapsoftware.com/Schemas/MadCap.xsd" />
</TocEntry>
```
This would make the **Perl Compatible Regular Expressions (PCRE**) chapter a subchapter of the **Regular expression (regexp) parser** chapter, on the same level as the **Options of regular expression parsers** subchapter.

Reply:
@gaborznagy 
/!\ Moving files from a folder to another folder has implications within our cross-reference/links system that can potentially affect up to hundreds of files. This is easily handled by our tool, but can break content if not handled from within the tool and requires special care and in-depth knowledge of what can go wrong. 

So yes, in general, we store reused chapters/subchapters in the `shared` folder, reused snippet chunks in the `shared/chunk` folder and reused notes, cautions, tips and warnings in the `shared/wnt` folder. 
However, **please do not move files around on your own**. The team is happy to help in this case to avoid breaking the code.

If you create a new file, however, feel free to add them to the abovementioned folders directly.

If you are interested in the "what can go wrong" part as well, let's arrange a call. ;)

Reply:
Thanks for the thorough help!

I would like to just replace the current content of the `flags()` option under `regexp-parser()`.
The current one uses `flags()` option that are available for source drivers, and I would like to insert the one which is mentioned in the "PCRE flags() option" chapter. Unfortunately (for me), PCRE flags() are not written as a chunk to insert.

I would like to avoid including it as a subchapter under *Options of Regular Expression parser*, as it is already a 2-level deep part:
```
Regular expression (regexp) parser
 |- Options of Regular expression parsers
```
I certainly don't want this layout:
```
Regular expression (regexp) parser
 |- Options of Regular expression parsers
     |- Perl Compatible Regular Expressions (PCRE) flags()
```
or to have it on the same level.
```
Regular expression (regexp) parser
 |- Options of Regular expression parsers
 |- Perl Compatible Regular Expressions (PCRE) flags()
```

It's just another option, it shouldn't be any more special than the other options, so it should be inlined on the same page.
In the end, I think the layout of the original *Regular Expressions* chapter should look like the `type()` documentation, see: https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.36/administration-guide/80#TOPIC-1768810
My idea is that instead of putting flags() contents into whole `.htm` files, I would restructure the PCRE `flags()` documentation into snippets.
Or something like that. I think we should arrange that call as even I can't follow my chain of thought.